consensus/poet/core/sawtooth_poet/poet_consensus/utils.py

```diff
@@ -142,54 +142,95 @@ def build_certificate_list(block_header,
     return list(certificates)
 
 
-def create_validator_state(validator_info,
-                           current_validator_state):
-    """Starting with the current validator state (or None), create validator
+def get_current_validator_state(validator_info,
+                                consensus_state,
+                                block_cache):
+    """Fetches the current validator state for the validator, creating it if
+    necessary.
+
+    Args:
+        validator_info (ValidatorInfo): The validator information
+            corresponding to the consensus state
+        consensus_state (ConsensusState): The consensus state from which
+            validator state should be fecthed/derived.
+        block_cache (BlockCache): The block store cache
+
+    Returns:
+        ValidatorState object representing the validator's state at the
+            point in time of the consensus state
+    """
+    # Fetch the validator state.  If it doesn't exist, then create an initial
+    # validator state object, including figuring out the block number in which
+    # the validator was most-recently registered.
+    validator_state = \
+        consensus_state.get_validator_state(
+            validator_id=validator_info.id)
+
+    if validator_state is None:
+        enclosing_block = \
+            block_cache.block_store.get_block_by_transaction_id(
+                validator_info.transaction_id)
+        validator_state = \
+            ValidatorState(
+                commit_block_number=enclosing_block.block_num,
+                key_block_claim_count=0,
+                poet_public_key=validator_info.signup_info.poet_public_key,
+                total_block_claim_count=0)
+
+    return validator_state
+
+
+def create_updated_validator_state(validator_info,
```

Comment:
I am okay with `next_validator_state` but reject `update_validator_state` as the latter implies updating in place. The reason the function name was prefixed with `create_` was to be explicit that it was non-destructive to existing/current validator state.  Would `create_next_validator_state` be acceptable?